trinity/nodes/base.py

```diff
@@ -77,7 +83,7 @@ def add_service(self, service: BaseService) -> None:
         else:
             self._auxiliary_services.append(service)
 
-    def make_ipc_server(self) -> IPCServer:
+    def make_ipc_server(self) -> Union[IPCServer, BaseService]:
```

Comment:
These two don't share a common base class at all, so `Union` is the only way to go if that API wants to either return `IPCServer` or `BaseService`. This is related to https://github.com/ethereum/py-evm/issues/689